src/modules/commander/commander.cpp

```diff
@@ -1237,46 +1237,54 @@ static void commander_set_home_position(orb_advert_t &homePub, home_position_s &
 					const vehicle_local_position_s &localPosition, const vehicle_global_position_s &globalPosition,
 					const vehicle_attitude_s &attitude)
 {
-	//Need global and local position fix to be able to set home
-	if (!status_flags.condition_global_position_valid || !status_flags.condition_local_position_valid) {
-		return;
-	}
+	// Need global and local position fix to be able to set home
+	// Ensure that the GPS accuracy is good enough
+	if (status_flags.condition_global_position_valid
+			&& status_flags.condition_local_position_valid
+			&& (globalPosition.eph > eph_threshold || globalPosition.epv > epv_threshold)) {
 
-	//Ensure that the GPS accuracy is good enough for intializing home
-	if (globalPosition.eph > eph_threshold || globalPosition.epv > epv_threshold) {
-		return;
-	}
+		//Set home position
+		home.timestamp = hrt_absolute_time();
+		home.lat = globalPosition.lat;
+		home.lon = globalPosition.lon;
+		home.valid_hpos = true;
+
+		home.alt = globalPosition.alt;
+		home.valid_alt = true;
 
-	//Set home position
-	home.timestamp = hrt_absolute_time();
-	home.lat = globalPosition.lat;
-	home.lon = globalPosition.lon;
-	home.alt = globalPosition.alt;
+		home.x = localPosition.x;
+		home.y = localPosition.y;
+		home.z = localPosition.z;
 
-	home.x = localPosition.x;
-	home.y = localPosition.y;
-	home.z = localPosition.z;
+		matrix::Eulerf euler = matrix::Quatf(attitude.q);
+		home.yaw = euler.psi();
 
-	matrix::Eulerf euler = matrix::Quatf(attitude.q);
-	home.yaw = euler.psi();
+		PX4_INFO("home: %.7f, %.7f, %.2f", home.lat, home.lon, (double)home.alt);
 
-	PX4_INFO("home: %.7f, %.7f, %.2f", home.lat, home.lon, (double)home.alt);
+		//Play tune first time we initialize HOME
+		if (!status_flags.condition_home_position_valid) {
+			tune_home_set(true);
+		}
+
+		/* mark home position as set */
+		status_flags.condition_home_position_valid = true;
+
+	} else if (!home.valid_alt && localPosition.z_global) {
+		// Handle case where we have started global height estimation after takeoff and can only set the home altitude
+		home.timestamp = hrt_absolute_time();
+		home.alt = localPosition.ref_alt;
+		home.valid_alt = true;
```

Comment:
Explicitly set `home.valid_hpos = false` for the sake of readability?

Reply:
What if it is already true because a horizontal position has been set? That can't happen at the moment, but it could in the future.

Reply:
That answers my real question. Thanks